src/twisted/internet/defer.py

```diff
@@ -167,6 +168,8 @@ def maybeDeferred(f, *args, **kw):
 
     if isinstance(result, Deferred):
         return result
+    elif iscoroutine(result):
```

Comment:
I think this is too dangerous in general, in a mixed asyncio/twisted application this could be a legitimate return value

Perhaps a warning would be better?